### PR TITLE
Chunk store test integrated

### PR DESCRIPTION
**Pull request by krishnaIndia**
Have implemented the initial utility functions and structured it. Will continue writing the helper functions and then the unit tests.

### CODE REVIEW

File: src/chunk_store/test.rs [hunk 1]
```diff
@@ -0,0 +1,82 @@
+extern crate maidsafe_types;
+extern crate rand;
+
+use rand::{thread_rng, Rng};
+use std::str;
+use chunk_store::ChunkStore;
+use std::path::Path;
+use std::fs::{remove_dir_all};
+
+static ONE_KB: u64 = 1024u64;
```
Comment (dirvine) [inline src/chunk_store/test.rs]: I think this can be just `static ONE_KB: u64 = 1024;`

File: src/chunk_store/test.rs [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+static AES_PADDING: u64 = 16u64;
```
Comment (dirvine) [inline src/chunk_store/test.rs]: We do no tneed to consider encrypting these chunks in the chunk-store any more. They will be session based so we can store them as they are (already encrypted anyway). Keeps it simpler. So this becomes literally a HashMap of <NameType, Data> . Should simplify it a lot :-)

**Comment by krishnaIndia** [comment]
Updated the unit tests. Two more unit tests are pending. Unused code to be removed after the tests are wired up. Implemented `get` in chunk_store.rs for writing the tests.

File: Cargo.toml [hunk 3]
```diff
@@ -25,3 +25,9 @@ git = "https://github.com/dirvine/maidsafe_types"
 
 [dependencies.routing]
 git = "https://github.com/dirvine/routing"
+
+[dependencies.self_encryption]
```
Comment (dirvine) [inline Cargo.toml]: I do not think we need self_encryption as a dependency here.

File: src/lib.rs [hunk 4]
```diff
@@ -10,6 +10,10 @@ extern crate sodiumoxide;
 extern crate rustc_serialize;
 extern crate cbor;
 extern crate time;
+extern crate bchannel;
```
Comment (dirvine) [inline src/lib.rs]: Not sure bchannel is required either.

File: Cargo.toml (hunk 3 above)
Comment (krishnaIndia) [inline Cargo.toml]: It was not compiling at my end. Will check one more time, removing the dependency

File: src/lib.rs (hunk 4 above)
Comment (krishnaIndia) [inline src/lib.rs]: I might have messed it while merging. I ll check this along with the self_encryption dependency.

**Comment by krishnaIndia** [comment]
Tests have been integrated. But there are three failing tests.

**Comment by krishnaIndia** [comment]
The failing tests are fixed. All tests are passing for ChunkStore

**Comment by dirvine** [comment]
Nice @krishnaIndia  Thanks

**Pull request merged**